Description: Ensure that octave.jar is built even if java is disabled
 On some archs we pass --disable-java to the configure script, but we still
 want octave-common to contain octave.jar if built on those archs.
Author: Sébastien Villemot <sebastien@debian.org>
Forwarded: not-needed
Reviewed-by: Rafael Laboissière <rafael@debian.org>
Last-Update: 2016-11-20
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/scripts/java/module.mk
+++ b/scripts/java/module.mk
@@ -1,3 +1,6 @@
+JAVAC=javac
+JAR=jar
+
 FCN_FILE_DIRS += scripts/java
 
 scripts_java_FCN_FILES = \
@@ -11,15 +14,12 @@ PKG_ADD_FILES += java/PKG_ADD
   scripts/java/javarmpath.m \
   scripts/java/usejava.m
 
-if AMCOND_HAVE_JAVA
 scripts_java_JAR_FILES = scripts/java/octave.jar
 
 JAR_FILES += $(scripts_java_JAR_FILES)
-endif
 
 org_octave_dir = org/octave
 
-if AMCOND_HAVE_JAVA
 JAVA_SRC = \
   $(org_octave_dir)/ClassHelper.java \
   $(org_octave_dir)/OctClassLoader.java \
@@ -26,7 +26,6 @@ JAVA_SRC = \
   $(org_octave_dir)/Octave.java \
   $(org_octave_dir)/OctaveReference.java \
   $(org_octave_dir)/Matrix.java
-endif
 
 JAVA_CLASSES = $(JAVA_SRC:.java=.class)
 
@@ -38,13 +38,11 @@ java/images.stamp: $(srcdir_java_JAVA_IM
 	$(AM_V_GEN)$(MKDIR_P) scripts/java/$(org_octave_dir) && \
 		( cd $(srcdir)/scripts/java; "$(JAVAC)" -source 1.3 -target 1.3 -Xlint:-options -d $(abs_top_builddir)/scripts/java $(org_octave_dir)/$(<F) )
 
-if AMCOND_HAVE_JAVA
 scripts/java/octave.jar: $(scripts_java_JAVA_CLASSES)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	( cd scripts/java; \
 	  "$(JAR)" cf octave.jar-t $(JAVA_CLASSES) ) && \
 	mv $@-t $@
-endif
 
 scripts_javadir = $(fcnfiledir)/java
 
